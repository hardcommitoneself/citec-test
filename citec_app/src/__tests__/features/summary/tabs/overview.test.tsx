// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'
import { BrowserRouter } from 'react-router-dom'
import { dataTest } from '__tests__/features/rebalance/backtest/dataTest'
import { mount } from 'enzyme'
import { Overview } from 'features/summary/tabs'

jest.mock('react-apexcharts', () => {
   return {
      __esModule: true,
      default: () => {
         return <div />
      },
   }
})
const analysis = {
   fundamentals: {
      total_assets: 4,
      total_positions: 4,
      pe: 73.739873,
      pb: 28.025954,
      dividend_yield: 0.517303,
   },
   risk: {
      sharpe: -1.002506,
      volatility: 0.068686,
      beta: 0.315348,
      alpha: -0.000072,
   },
   returns: {
      returns_1M: -0.0148,
      returns_3M: -0.0103,
      returns_6M: null,
      CAGR: -0.0575,
   },
   capital_gains: {
      outstanding_balance: 1310000,
      unrealized_cap_losses: 101000,
      unrealized_cap_gains: 11234,
      realized_cap_losses: 0,
      realized_cap_gains: 0,
      taxes_cost: 0,
      taxes_rate: 0.4,
   },
   sectors: {
      sector: ['Cash', 'Consumer Discretionary', 'Information Technology'],
      percentage: [0.763359, 0.007634, 0.229008],
      percentage_benchmark: [0.0706, 0.1086],
   },
   factors: {
      factor: [
         'style',
         'size',
         'yield',
         'quality',
         'volatility',
         'momentum',
         'liquidity',
      ],
      percentage: [
         0.513052, 0.618264, 0.59449, 0.5672, 0.432629, 0.529742, 0.465484,
      ],
   },
   segment_matrix: {
      style: ['Val', 'Val', 'Val', 'Bld', 'Bld', 'Bld', 'Gwth', 'Gwth', 'Gwth'],
      size: [
         'Large',
         'Mid',
         'Small',
         'Large',
         'Mid',
         'Small',
         'Large',
         'Mid',
         'Small',
      ],
      weight: [0, 0, 0, 19.083969, 0, 0, 4.580153, 0, 0],
   },
   top_holdings: {
      ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
      weight: [0.763359, 0.19084, 0.038168, 0.007634],
      outstanding_balance: [1000000, 250000, 50000, 10000],
      unrealized_capital_gains: [0, -65000, 11234, -36000],
   },
   concentration: {
      ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
      concentration: [76.335878, 95.419847, 99.236641, 100],
   },
   var: {
      quantile: [
         5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95,
      ],
      value: [
         0, 0, 0, 0, 0.010967, 0.021321, 0.030917, 0.040021, 0.04883, 0.0575,
         0.066169, 0.074978, 0.084083, 0.093678, 0.104032, 0.115563, 0.129003,
         0.145913, 0.170978,
      ],
   },
   risk_returns: {
      ticker: ['portfolio', 'benchmark'],
      returns: [-0.0575, -0.122369],
      volatility: [0.068686, 0.190293],
      sharpe: [0.837133, 0.643053],
   },
}

const mockStore = configureStore()
const store = mockStore(exampleReduxState)

describe('Portfolio Analysis tab OverView', () => {
   test('renders Portfolio Analysis page', () => {
      const { container } = render(
         <Provider store={store}>
            <BrowserRouter>
               <Overview backtest={dataTest} analysis={analysis} />
            </BrowserRouter>
         </Provider>
      )
      expect(container).toMatchSnapshot()
   })
})

describe('Portfolio Analysis tab OverView SWAL', () => {
   test('calls Swal Sector', () => {
      const { getByTestId } = render(
         <Provider store={store}>
            <BrowserRouter>
               <Overview backtest={dataTest} analysis={analysis} />
            </BrowserRouter>
         </Provider>
      )

      const button = getByTestId('info-summary-overview-sectors')
      expect(button).toBeTruthy()
   })

   it('click on information icons', () => {
      const component = mount(
         <Provider store={store}>
            <BrowserRouter>
               <Overview backtest={dataTest} analysis={analysis} />
            </BrowserRouter>
         </Provider>
      )
      component.find({ 'data-testid': 'cagralert-test' }).simulate('click')
      component.find({ 'data-testid': 'priceperbookalert-test' }).simulate('click')
      component.find({ 'data-testid': 'betaalert-test' }).simulate('click')
      component.find({ 'data-testid': 'alfaalert-test' }).simulate('click')
      component
         .find({ 'data-testid': 'trackingdifferencealert-test' })
         .simulate('click')
      component.find({ 'data-testid': 'rsquarealert-test' }).simulate('click')

      component
         .find({ 'data-testid': 'info-summary-overview-segment' })
         .simulate('click')
      component
         .find({ 'data-testid': 'info-summary-overview-factor' })
         .simulate('click')
   })
})
