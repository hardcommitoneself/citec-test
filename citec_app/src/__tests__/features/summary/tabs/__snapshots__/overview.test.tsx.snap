// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Analysis tab OverView renders Portfolio Analysis page 1`] = `
<div>
  <div
    class="grid grid-cols-4 gap-6 mt-2"
  >
    <div
      class="CustomizationBox bg-inherit"
    >
      <img
        alt=""
        src="investment.png"
      />
      <div
        class="presentation"
      >
        <p>
          $ 1,050,000
        </p>
        <h1>
          Investment
        </h1>
        <p>
          <span>
            No. of Positions 
          </span>
          <span
            style="margin-left: auto;"
          >
            3
          </span>
        </p>
      </div>
    </div>
    <div
      class="CustomizationBox bg-inherit"
    >
      <button
        class="moreinfo info-totals text-[#0BACA9]"
      >
        <svg
          fill="currentColor"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
          />
          <path
            d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
          />
        </svg>
      </button>
      <img
        alt=""
        src="cap.png"
      />
      <div
        class="presentation"
      >
        <p>
          -$ NaN
        </p>
        <h1>
          Avg. market cap
        </h1>
      </div>
    </div>
    <div
      class="CustomizationBox bg-inherit"
    >
      <button
        class="moreinfo info-totals text-[#0BACA9]"
      >
        <svg
          fill="currentColor"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
          />
          <path
            d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
          />
        </svg>
      </button>
      <img
        alt=""
        src="price.png"
      />
      <div
        class="presentation"
      >
        <p>
          24.32
        </p>
        <h1>
          Price per earnings
        </h1>
      </div>
    </div>
    <div
      class="CustomizationBox bg-inherit"
    >
      <button
        class="moreinfo info-totals text-[#0BACA9]"
      >
        <svg
          fill="currentColor"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
          />
          <path
            d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
          />
        </svg>
      </button>
      <img
        alt=""
        src="overview_percentage.png"
      />
      <div
        class="presentation"
      >
        <p>
          0.59%
        </p>
        <h1>
          Dividend yield
        </h1>
      </div>
    </div>
  </div>
  <div
    class="RiskReturnGrid"
  >
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1>
          Sector composition
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-overview-sectors"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class="chartRiskReturn"
      >
        <div
          class=""
        >
          <div
            class="text-[#6C8394] text-sm"
          >
            <div
              class="flex justify-center items-center"
            >
              <div
                class="h-3 w-3 rounded-sm bg-[#21BAB7]"
              />
              <span
                class="ml-2"
              >
                Porftolio
              </span>
              <div
                class="ml-2 h-3 w-3 rounded-sm bg-[#3CB4EE]"
              />
              <span
                class="ml-2"
              >
                Benchmark
              </span>
            </div>
          </div>
          <div
            class="-mt-12"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      class="box_chart_area KeyMetricsTableAreaWrapper"
    >
      <div
        class="top_area !text-[14px]"
      >
        <h1>
          Key metrics
        </h1>
        <table
          class="KeyMetricsTableArea"
        >
          <thead>
            <tr>
              <th />
              <th>
                Portfolio
              </th>
              <th>
                Benchmark
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Return (3 months)
              </td>
              <td>
                -0.37%
              </td>
              <td>
                NaN%
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Return (6 months)
              </td>
              <td>
                -0.28%
              </td>
              <td>
                NaN%
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                CAGR
                <button
                  class="mx-2"
                  data-testid="cagralert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                -10.58%
              </td>
              <td>
                NaN%
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Prices per earnings
              </td>
              <td>
                24.32
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Price per book value
                <button
                  class="mx-2"
                  data-testid="priceperbookalert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                38.96
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Dividend yield
              </td>
              <td>
                0.59%
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Beta
                <button
                  class="mx-2"
                  data-testid="betaalert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                1.04
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Alpha
                <button
                  class="mx-2"
                  data-testid="alfaalert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                0.00%
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                Tracking difference
                <button
                  class="mx-2"
                  data-testid="trackingdifferencealert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                0.01%
              </td>
              <td>
                N.A
              </td>
            </tr>
            <tr>
              <td
                class="keyWhithInfo p-2"
              >
                R square
                <button
                  class="mx-2"
                  data-testid="rsquarealert-test"
                >
                  <svg
                    fill="currentColor"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.83918 1.39157C5.38809 1.39157 3.99643 1.97801 2.97036 3.02189C1.94428 4.06576 1.36784 5.48156 1.36784 6.95783C1.36784 8.43409 1.94428 9.84989 2.97036 10.8938C3.99643 11.9376 5.38809 12.5241 6.83918 12.5241C8.29027 12.5241 9.68193 11.9376 10.708 10.8938C11.7341 9.84989 12.3105 8.43409 12.3105 6.95783C12.3105 5.48156 11.7341 4.06576 10.708 3.02189C9.68193 1.97801 8.29027 1.39157 6.83918 1.39157ZM0 6.95783C0 3.11502 3.0619 0 6.83918 0C10.6165 0 13.6784 3.11502 13.6784 6.95783C13.6784 10.8006 10.6165 13.9157 6.83918 13.9157C3.0619 13.9157 0 10.8006 0 6.95783Z"
                    />
                    <path
                      d="M6.83838 5.56633C7.01976 5.56633 7.19372 5.63964 7.32198 5.77012C7.45024 5.90061 7.5223 6.07758 7.5223 6.26212V10.4368C7.5223 10.6213 7.45024 10.7983 7.32198 10.9288C7.19372 11.0593 7.01976 11.1326 6.83838 11.1326C6.65699 11.1326 6.48303 11.0593 6.35477 10.9288C6.22651 10.7983 6.15446 10.6213 6.15446 10.4368V6.26212C6.15446 6.07758 6.22651 5.90061 6.35477 5.77012C6.48303 5.63964 6.65699 5.56633 6.83838 5.56633ZM7.86425 3.82688C7.86425 4.10368 7.75617 4.36914 7.56378 4.56487C7.37139 4.76059 7.11046 4.87055 6.83838 4.87055C6.5663 4.87055 6.30536 4.76059 6.11297 4.56487C5.92058 4.36914 5.8125 4.10368 5.8125 3.82688C5.8125 3.55008 5.92058 3.28461 6.11297 3.08889C6.30536 2.89316 6.5663 2.7832 6.83838 2.7832C7.11046 2.7832 7.37139 2.89316 7.56378 3.08889C7.75617 3.28461 7.86425 3.55008 7.86425 3.82688Z"
                    />
                  </svg>
                </button>
              </td>
              <td>
                99.64%
              </td>
              <td>
                N.A
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1>
          Segment matrix
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-overview-segment"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class="chartRiskReturn"
      >
        <div />
      </div>
    </div>
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1>
          Factors
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-overview-factor"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class="chartRiskReturn"
      >
        <div />
      </div>
    </div>
  </div>
  <div
    class="BacktestChartWrapper"
  >
    <div
      class="rowBack"
    >
      <p>
        Tracking Error probability distribution
      </p>
      <img
        alt=""
        class="moreinfo"
        data-testid="tracking-error-test"
        src="group.png"
      />
    </div>
    <div
      class="rowBack"
    >
      <h1>
        Tracking Error: 
      </h1>
      <div
        class="tracking_error_green"
      >
        1.3
         
        %
      </div>
    </div>
    <div
      class="pt-3"
    >
      <div />
    </div>
  </div>
</div>
`;
